azurelinuxagent/ga/monitor.py

```diff
@@ -177,11 +180,35 @@ def collect_and_send_events(self):
         except ProtocolError as e:
             logger.error("{0}", e)
 
+    def send_daily_heartbeat(self):
+        """
+        Send the daily heartbeat containing the installed and current agent versions.
+        Send once during the first 30 minutes of the day.
+        """
+        if datetime.datetime.time() <= datetime.time(0, minute=30):
+            try:
+                msg = u"{0}:{1}".format(
+                    INSTALLED_VERSION,
+                    CURRENT_VERSION)
+                logger.info(msg)
+                add_event(
+                    AGENT_NAME,
+                    version=CURRENT_VERSION,
+                    op=WALAEventOperation.DailyHeartBeat,
+                    is_success=True,
+                    message=msg)
+            except Exception as e:
+                logger.warn("Exception sending daily heartbeat on {0}: {1}",
+                    today.strftime(ISOFORMAT),
+                    e)
+
     def daemon(self):
         last_heartbeat = datetime.datetime.min
         period = datetime.timedelta(minutes=30)
         while True:
             if (datetime.datetime.now() - last_heartbeat) > period:
+                send_daily_heartbeat()
```

Comment:
Wouldn't it be better to send the daily heartbeat once and then wait for a 24 hour period?